### ISSUE TITLE
False positive on class-string annotation for string property

### ISSUE THREAD

**Issue by tjveldhuizen** [phpstan/phpstan-doctrine#162]
In my project, I need to save the fully qualified class name of an object in one of my entities. To be able to call static methods on that class, I think it's required to annotate the property with `@var class-string`. However phpstan-doctrine complains:

`Property MyObject::$myClassName type mapping mismatch: database can contain string but property expects class-string<MyInterface>.`

I am not sure if something is wrong with my annotation, or this is a bug in phpstan-doctrine. I also tried to use `@var class-string<MyInterface>|string`, which removes the mentioned error, but is less strict and does not give autocompletion on the methods in the interface, in PhpStorm.

```php
interface MyInterface {
    public static method doSomething(): void;
}

/**
 * @ORM\Entity
 */
class MyObject
{
    /**
     * @var class-string<MyInterface>
     * @ORM\Column
     */
    public string $myClassName;
}

$object = new MyObject();
$class = $object->myClassName;
$class::doSomething();
```

**Comment by ondrejmirtes** [comment]
You need a custom Doctrine type that actually guarantees you it will only ever put `class-string` value into your properties.

This README section is relevant to that: https://github.com/phpstan/phpstan-doctrine#reflectiondescriptor (you need the right typehints in `convertToDatabaseValue` and `convertToPHPValue` methods).
